modules/nextflow/src/main/groovy/nextflow/util/NullablePath.groovy

```diff
@@ -0,0 +1,31 @@
+package nextflow.util
+
+
+/**
+ * @author : jorge <jorge.aguilera@seqera.io>
+ *
+ */
+class NullablePath implements CharSequence{
```

Comment:
Have you explored the possibility to have `NullablePath implements Path` and delegate the implementation to a fack local path object 

For example https://github.com/nextflow-io/nextflow/blob/c70eb12d1caac1ca3e1acb7c5b2a0adbfae18ebe/modules/nextflow/src/main/groovy/nextflow/scm/ProviderPath.groovy#L45-L52

Reply:
yes, I've considered it and in this way include NullablePathS to the list of files produced for the task

On the one hand, the code looks great and is simple but I was not sure if we wanted to have this kind of file because the user will need to check if it exists

Reply:
I think if we go for this approach should be consistent with the expected type, `Path` in this case.

Reply:
Ok, I'll change to Path